encoding/av1-roi-calculator/js/av1-roi-calculator.js

```diff
@@ -0,0 +1,156 @@
+   */
+  function roundDecimals(n, decimalPlaces) {
+    var diff = Math.pow(10, decimalPlaces);
+    return Math.round((n.valueOf() + Number.EPSILON) * diff) / diff;
+  }
+
+  /**
+   * Generic handler that sets 'is-invalid' class to input, if value is NaN or <0
+   * @param onInputValue
+   * @returns {function(value : Number | NaN): (undefined)}
+   */
+  function inputEventHandler(onInputValue) {
+    return function(e) {
+      var inputEl = $(this);
+      inputEl.removeClass('is-invalid');
+
+      var numberValue = Number(e.target.value);
+      if (isNaN(numberValue) || numberValue < 0
+        || countDecimals(numberValue) > 2) {
+        inputEl.addClass('is-invalid');
+        onInputValue(NaN);
+        return;
+      }
+
+      onInputValue(numberValue);
+    };
+  }
+
+  (function initAv1Form() {
+    // form values for calculation
+    let encodingCostPerMinute = 0.02;
+    let ingressCostPerGb = 0.00;
+    let egressCostPerGb = 0.04;
+
+    let numberOfStreamsUhd = 2;
+    let numberOfStreamsHd = 3;
+    let numberOfStreamsSd = 4;
+
+    const multiplierStreamUhd = 4;
+    const multiplierStreamHd = 2;
+    const multiplierStreamSd = 1;
+    const multiplierTechPerTitle = 1.1;
+    const multiplierTech3Pass = 2; // Multipass
+    const multiplierCodecAv1 = 10;
+
+    // streams/renditions
+    const improvementsAv1H264 = 0.5; // 50%
+    const improvementsAv1H265 = 0.7; // 30%
+    const mbpsH264Uhd = 12;
+    const mbpsH264Hd = 5;
+    const mbpsH264Sd = 1;
+    const uhdRenditionGbPerMinH264 = (mbpsH264Uhd * 60) / (8 * 1000); // GB (8 * 1000) per min (60 seconds)
+    const uhdRenditionGbPerMinH265 = uhdRenditionGbPerMinH264 * improvementsAv1H264 / improvementsAv1H265;
+    const uhdRenditionGbPerMinAv1 = uhdRenditionGbPerMinH264 * improvementsAv1H264;
+    
+    // result elements
+    const av1H264Element = $('#av1-h264');
+    const av1H265Element = $('#av1-h265');
+
+    const calculateBreakEvenPoints = () => {
+      // stream composition
+      const multiplierStreamComposition = numberOfStreamsUhd * multiplierStreamUhd
+        + numberOfStreamsHd * multiplierStreamHd
+        + numberOfStreamsSd * multiplierStreamSd;
+      const encodingCostPerMinuteAv1 = encodingCostPerMinute * multiplierStreamComposition
+        * multiplierTech3Pass * multiplierTechPerTitle * multiplierCodecAv1;
+
+      // all stream bandwidth
+      const multiplierStreamCompositionMbps = numberOfStreamsUhd * mbpsH264Uhd
+        + numberOfStreamsHd * mbpsH264Hd
+        + numberOfStreamsSd * mbpsH264Sd;
+      const allRenditionsGbPerMinH264 = (multiplierStreamCompositionMbps * 60) / (8 * 1000); // GB (8 * 1000) per min (60 seconds)
+      const allRenditionsGbPerMinAv1 = allRenditionsGbPerMinH264 * improvementsAv1H264;
+      
+      // final result
+      const av1H264BreakEven = (encodingCostPerMinuteAv1 + allRenditionsGbPerMinAv1 * ingressCostPerGb)
+        / (egressCostPerGb * (uhdRenditionGbPerMinH264 - uhdRenditionGbPerMinAv1));
```

Comment:
we could split this up into 3 steps to make it more clear:
- calculate `oneTimeCosts`
- calculate `savingsPerView`
- divide them

Reply:
Done in [09b786a](https://github.com/bitmovin/demos/pull/92/commits/09b786a89c475d84de1e8762d5b33095dbc9c229)